src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Handlers/RetryHandler.cs

```diff
@@ -56,15 +56,15 @@ public class RetryHandler : DelegatingHandler
         protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
             HttpResponseMessage response = null;
+
             for (int attempt = 0; attempt < RetryHandler.MaxAttempts; ++attempt)
             {
                 try
                 {
-                    response = await base
-                        .SendAsync(request: request, cancellationToken: cancellationToken)
+                    response = await base.SendAsync(request: request, cancellationToken: cancellationToken)
                         .ConfigureAwait(continueOnCapturedContext: false);
-
-                    if (attempt == RetryHandler.MaxAttempts ||
+                    
+                    if (attempt == RetryHandler.MaxAttempts - 1 ||
```

Comment:
Previously the condition was "attempt == RetryHandler.MaxAttempts", which would never be true. I think the point here is directly return the response after MaxAttempts of tries, and don't do the rest of the stuff.